[AY1920S1-CS2113-F09-2] Cube

kuromono opened the pull request:
@LL-Pengfei
@ZKathrynx
@tygq13
@parvathi14
@kuromono

LL-Pengfei commented:
Hahaha we are done this week.

Rishav1 commented on src/main/java/cube/MainApp.java:
```diff
@@ -0,0 +1,45 @@
+    public void start(Stage stage) {
+        init("data");
+        //stage.setMinHeight(configStorage.getWindowHeight());
+        //stage.setMinWidth(configStorage.getWindowWidth());
```
Avoid commented codes.

Rishav1 commented on src/main/java/cube/logic/command/Command.java:
```diff
@@ -0,0 +1,14 @@
+		return false;
+	}
+
+	public abstract CommandResult execute(FoodList list, StorageManager storage) throws CommandException;
```
Avoid one line functions.

Rishav1 commented on src/main/java/cube/logic/command/CommandResult.java:
```diff
@@ -0,0 +1,68 @@
+                && showHelp == otherCommandResult.showHelp
+                && exit == otherCommandResult.exit;
+    }
+    */
```
Bad.

Rishav1 commented on src/main/java/cube/logic/command/DeleteCommand.java:
```diff
@@ -0,0 +1,132 @@
+	 */
+	private void checkValidName(FoodList list) throws CommandException {
+		if (!list.existsName(deleteDescription)) {
+			throw new CommandException(CommandErrorMessage.FOOD_NOT_EXISTS);
```
Incorrect indentation.

Rishav1 commented on src/main/java/cube/logic/command/DeleteCommand.java:
```diff
@@ -0,0 +1,132 @@
+	private void checkValidType(FoodList list) throws CommandException {
+		if (!list.existsType(deleteDescription)) {
+			throw new CommandException(CommandErrorMessage.FOOD_NOT_EXISTS);
+		}
```
Incorrect indentation.

Rishav1 commented on src/main/java/cube/logic/command/DeleteCommand.java:
```diff
@@ -0,0 +1,132 @@
+				storage.storeFoodList(list);
+				return new CommandResult(String.format(MESSAGE_SUCCESS_MULTIPLE, deleteDescription, count, list.size()));
+		}
+		return null;
```
Problem with indentation. Follow coding standards for the course.

Rishav1 commented on src/main/java/cube/logic/command/ExitCommand.java:
```diff
@@ -0,0 +1,39 @@
+	 */
+	@Override
+	public CommandResult execute(FoodList list, StorageManager storage) {
+		return new CommandResult(MESSAGE_SUCCESS, false, true);
```
Indentation issue.

Rishav1 commented on src/main/java/cube/logic/command/FindCommand.java:
```diff
@@ -0,0 +1,123 @@
+                    result.sort(sortType);
+                }
+                return new CommandResult(String.format(MESSAGE_SUCCESS_MULTIPLE, count,result,list.size()));
+        }
```
Switch case is not indented as per the coding standard for the course. Also violation of SLAP.

Rishav1 commented on src/main/java/cube/logic/command/GenerateRevenueCommand.java:
```diff
@@ -0,0 +1,132 @@
+     * @param GenerateRevenueDescription The food name or food type whose revenue is to be generated.
+     * @param param The parameter is used to specify the type of generating revenue.
+     */
+    public GenerateRevenueCommand(String GenerateRevenueDescription, String param) {
```
Parameter `GenerateRevenueDescription` doesn't follow the naming standard.

Rishav1 commented on src/main/java/cube/logic/command/HelpCommand.java:
```diff
@@ -0,0 +1,33 @@
+
+public class HelpCommand extends Command{
+
+	String MESSAGE_SUCCESS = "Currently we support the following commands: (all command ignore cases)\n"
```
Indentation issue.

Rishav1 commented on src/main/java/cube/logic/command/ListCommand.java:
```diff
@@ -0,0 +1,29 @@
+import cube.model.Food;
+import cube.storage.StorageManager;
+
+public class ListCommand extends Command{
```
Indentation issue. Tabs are used instead of 4 spaces.

Rishav1 commented on src/main/java/cube/logic/command/ReminderCommand.java:
```diff
@@ -0,0 +1,68 @@
+
+public class ReminderCommand extends Command{
+
+    private String MESSAGE_SUCCESS = "";
```
Not following naming standard.

Rishav1 commented on src/main/java/cube/logic/command/ReminderCommand.java:
```diff
@@ -0,0 +1,68 @@
+            if (stock != 0 && stock < 5 ) {
+                stockReminder.add(food);
+            }
+        }
```
Violation of SLAP.

Rishav1 commented on src/main/java/cube/logic/parser/AddCommandParser.java:
```diff
@@ -0,0 +1,53 @@
+			if (args[i].equals("-t")) {
+				foodTypeIndex = i;
+			}
+			if (args[i].equals("-p")) {
```
Violation of SLAP.

Rishav1 commented on src/main/java/cube/logic/parser/FindCommandParser.java:
```diff
@@ -0,0 +1,37 @@
+                    if (args[i].equals("-sort")) {
+                        sortIndex = i;
+                    }
+                }
```
Violation of SLAP.

Rishav1 commented on src/main/java/cube/logic/parser/ListCommandParser.java:
```diff
@@ -0,0 +1,19 @@
+public class ListCommandParser implements ParserPrototype<ListCommand> {
+
+	public ListCommand parse(String[] args) throws ParserException {
+		if (args.length == 1) {
```
Indentation issue.

Rishav1 commented on src/main/java/cube/logic/parser/Parser.java:
```diff
@@ -0,0 +1,64 @@
+	 * @param fullCommand the command that user inputs.
+	 * @return corresponding command.
+	 */
+	public static Command parse (String fullCommand) throws ParserException {
```
Indentation issue.

Rishav1 reviewed (commented):
Good work team! Added a few comments in my review. Apart from indentation issues, the code is very well structured. Nice Work!

tygq13 committed:
bug fix: checks negative date

tygq13 committed:
bug fix: add an alr expired food

tygq13 committed:
Merge pull request #238 from tygq13/hotfix

Hotfix

kuromono committed:
Completed GUI layout draft (more or less, no styling)

kuromono committed:
Tidy up UI code slightly

kuromono committed:
Merge remote-tracking branch 'cube-upstream/master' into cube-gui

kuromono committed:
Merge pull request #239 from kuromono/cube-gui

Completed GUI layout draft (more or less, no styling)

kuromono committed:
BatchCommand : Fix not to import file not found

kuromono committed:
Merge pull request #240 from kuromono/cube-fix-bug

BatchCommand : Fix not to import file not found

kuromono committed:
ConfigCommand enum ignore case

kuromono committed:
Modifed FoodList views

kuromono committed:
Added statusbar views

kuromono committed:
Modified Main Views

kuromono committed:
Added UI styling CSS

kuromono committed:
Merge pull request #241 from kuromono/cube-gui

Finalise GUI layout & include styling

kuromono committed:
Update Ui.png

kuromono committed:
Merge pull request #242 from kuromono/cube-gui

Update Ui.png

LL-Pengfei committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-F09-2/main

# Conflicts:
#	src/main/java/cube/logic/command/HelpCommand.java

LL-Pengfei committed:
profit command not functional yet, to be pr-ed as profit command parser give the idea how it works

LL-Pengfei committed:
Merge pull request #243 from LL-Pengfei/master

profit command not fully functional yet, pushed in as the programme can at least run without crashing for now

kuromono committed:
Improve GUI layout

kuromono committed:
Refactor MainApp to CubeApp

kuromono committed:
Added a launcher

Default will launch GUI
or add cli parameter behind to launch CLI:
java -jar "Cube...jar" cli

kuromono committed:
Fix CubeLauncher & upgraded gradlew to 5.6.4

kuromono committed:
Merge remote-tracking branch 'cube-upstream/master' into cube-gui

kuromono committed:
Update Ui.png to latest GUI revision

kuromono committed:
Merge pull request #244 from kuromono/cube-gui

Updated Cube GUI & added launcher to GUI

LL-Pengfei committed:
uncommented list command test

LL-Pengfei committed:
Merge pull request #245 from LL-Pengfei/master

uncommented list command test

ZKathrynx committed:
Merge pull request #17 from AY1920S1-CS2113T-F09-2/master

sync

ZKathrynx committed:
fix parser sort type bug and parse double bug

ZKathrynx committed:
fixed food name and type lowercase and upper case

ZKathrynx committed:
fixed conflict

ZKathrynx committed:
Merge pull request #246 from ZKathrynx/master

Fixed parser bug #1

ZKathrynx committed:
fixed date bug and delete type bug

ZKathrynx committed:
Merge pull request #247 from ZKathrynx/master

fixed date bug and delete type bug

ZKathrynx committed:
allow finding partial keyword

ZKathrynx committed:
fixed empty field value bug

ZKathrynx committed:
Merge pull request #248 from ZKathrynx/master

Allow finding partial keyword and parser bug fix #2

kuromono committed:
Commented out unused getFoodList in ListCommandTest

- Causing gradle build to fail

kuromono committed:
Merge pull request #249 from kuromono/master

Commented out unused getFoodList in ListCommandTest

ZKathrynx committed:
Merge pull request #18 from AY1920S1-CS2113T-F09-2/master

Sync

kuromono committed:
Change GUI edit button to sell

kuromono committed:
FoodListCard : Handle NullPointer for some food fields

kuromono committed:
Temp fix: Catches all exceptions for command box

kuromono committed:
Refactor CubeLoadingException to be under Util -> Exception

kuromono committed:
Merge pull request #250 from kuromono/cube-gui

Modified GUI, added some exception catch for GUI

kuromono committed:
Removed e.printstacktrace from FielUtils

kuromono committed:
Fix config crash on invalid enum

parvathi14 committed:
Merge pull request #11 from AY1920S1-CS2113T-F09-2/master

Updating personal repo

Commit:
Minor changes, commiting to allow for pull request

kuromono committed:
Fix config command issues

Commit:
Merge branch 'master' of https://github.com/parvathi14/main

# Conflicts:
#	src/main/java/cube/logic/command/exception/CommandErrorMessage.java
#	src/main/java/cube/logic/command/util/CommandUtil.java

kuromono committed:
Merge pull request #253 from kuromono/cube-gui

Fix some config bugs

kuromono committed:
Batch : Supports generating empty template

kuromono committed:
Batch : Fix error in arraylist declaration

kuromono committed:
Fix Command Parser Null error

kuromono committed:
Remove e.printstacktrace in UiManager

kuromono committed:
Merge pull request #255 from kuromono/cube-fix-bug

Fix Command Parser Null Bug + BatchCommand generate empty file

kuromono committed:
Remove Gradle JavaFX plugin to make universal

tygq13 committed:
edit on help command

tygq13 committed:
support sort by price

tygq13 committed:
restruture help message

tygq13 committed:
Merge pull request #257 from tygq13/hotfix

support sort by price

Commit:
mid v1.4 List promotion command

parvathi14 committed:
Merge branch 'master' into master

parvathi14 committed:
Merge pull request #12 from AY1920S1-CS2113T-F09-2/master

Updating personal repo

Commit:
Merge branch 'master' of https://github.com/parvathi14/main

parvathi14 committed:
Merge pull request #259 from parvathi14/master

mid v1.4 List promotion feature

kuromono committed:
Change GUI buttons to Edit, Sell

kuromono committed:
FoodListCard : Expiry Date Formatted

kuromono committed:
Sets batch CSV to output in DD/MM/YYYY format

kuromono committed:
Merge remote-tracking branch 'cube-upstream/master' into cube-fix-bug

kuromono committed:
Update Ui.png

kuromono committed:
Merge pull request #261 from kuromono/cube-fix-bug

Batch CSV input/output & GUI DD/MM/YYYY format

Commit:
v1.4: Delete promotion and list promotion features

parvathi14 committed:
Merge pull request #262 from parvathi14/master

v1.4: Delete promotion and list promotion feature

Commit:
v1.4: Error handling of Promotion Command, fixed promotion function for Sold Command and other minor changes

parvathi14 committed:
Merge pull request #263 from parvathi14/master

v1.4: Error handling of Promotion Command, fixed promotion function f…

parvathi14 committed:
Merge pull request #13 from AY1920S1-CS2113T-F09-2/master

Updating personal repo

Commit:
Merge branch 'master' of https://github.com/parvathi14/main

kuromono committed:
Change Time Zone to GMT+8 (previously GMT-8), add getter for timezone

kuromono committed:
FileUtilCSV : Now checks Timezone for imports

kuromono committed:
Tests : Added BatchCommandTest

kuromono committed:
Added default cases for Batch & Config Command

kuromono committed:
Change BatchCommand constructor args to directly use enum

kuromono committed:
Test : Added ConfigCommandTest

kuromono committed:
Modified launcher not to launch to CLI

kuromono committed:
Merge pull request #264 from kuromono/cube-junit

Added JUnit tests for Batch, Config command

LL-Pengfei committed:
intermediate update, profitcommandparser done, profitcommand not quite finished /20191109 22:25

kuromono committed:
Refactor foodlist generation for testing to another class

kuromono committed:
Test : Add FileUtilCSV test

kuromono committed:
Test : Add FileUtilJSON test

kuromono committed:
Test : Add FileUtilJson test fixed

kuromono committed:
Test : Add StorageManager Test - for FoodStorage & ConfigStorage

kuromono committed:
Merge pull request #265 from kuromono/cube-junit

Add FileUtil CSV, Json test, add partial StorageManager test

LL-Pengfei committed:
profit command 70% completed /20191109 23:05

LL-Pengfei committed:
Merge pull request #266 from LL-Pengfei/master

profit command 70% done

LL-Pengfei committed:
profitcommand 90% done /20191110

LL-Pengfei committed:
Merge pull request #267 from LL-Pengfei/master

profitcommand 90% done /20191110

LL-Pengfei committed:
some code cleaning, wrapped around text, some javadoc comments /20191110 1:06am

LL-Pengfei committed:
more javadoc coments & code cleaning /20191110 /1:15

LL-Pengfei committed:
Merge pull request #268 from LL-Pengfei/master

profit command done, along with some javadoc comments and code cleaning for other parts

Commit:
JUnit testing for promotions and reminders (reminder test needs to be worked on)

parvathi14 committed:
Merge pull request #269 from parvathi14/master

JUnit test files for promotions and reminders

LL-Pengfei committed:
more code cleaning /20191110

LL-Pengfei committed:
Merge pull request #270 from LL-Pengfei/master

more code cleaning /20191110

LL-Pengfei committed:
bug fix /20191110

LL-Pengfei committed:
Merge pull request #271 from LL-Pengfei/master

bug fix for profitcommand (impt fix)/20191110

ZKathrynx committed:
Merge pull request #19 from AY1920S1-CS2113T-F09-2/master

Sync

LL-Pengfei committed:
added junit test for food /20191110 16:10

LL-Pengfei committed:
Merge pull request #272 from LL-Pengfei/master

added junit test for food /20191110 16:10

ZKathrynx committed:
implement promotion list

ZKathrynx committed:
Merge pull request #273 from ZKathrynx/master

Promotion list

ZKathrynx committed:
allow same food to have diff promo in diff time

ZKathrynx committed:
Merge pull request #274 from ZKathrynx/master

allow same food to have diff promo in diff time

LL-Pengfei committed:
junit for food completed /20191110

LL-Pengfei committed:
Merge pull request #275 from LL-Pengfei/master

junit for food completed /20191110

LL-Pengfei committed:
formatting change

LL-Pengfei committed:
junit test update /20191110

ZKathrynx committed:
add java doc

LL-Pengfei committed:
profit command junit test done /20191110

LL-Pengfei committed:
Merge pull request #276 from LL-Pengfei/master

profit command testing done /20191110

kuromono committed:
Remove RevenueCommand & Relevant stuff

kuromono committed:
Remove Food.getRevenue()

kuromono committed:
ProfitCommand : Add static generateAnnualProfitRevenue for GUI use

kuromono committed:
SoldCommand : Remove storerevenue to storage.

kuromono committed:
Update GUI to support new profit/revenue using generateAnnualProfitRevenue

kuromono committed:
Food : Initialise all variables to 0, fix expirydate can be null. (exception due to simpledateformat)

kuromono committed:
Fix GUI

kuromono committed:
Merge pull request #277 from kuromono/cube-fix-bug

Removed Revenue Command & Update GUI to new Revenue/Profit

LL-Pengfei committed:
corresponding changes in food test (remove total revenue generation)

LL-Pengfei committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-F09-2/main

LL-Pengfei committed:
corresponding changes to foodtest due to change in food

ZKathrynx committed:
checkstyle #1

LL-Pengfei committed:
profit command javadoc comment added

ZKathrynx committed:
solve conflicts

ZKathrynx committed:
Merge pull request #278 from ZKathrynx/master

Checkstyle #1

LL-Pengfei committed:
profit storage code cleaned /20191110

LL-Pengfei committed:
Merge pull request #279 from LL-Pengfei/master

assorted code cleaning /20191110

kuromono committed:
Update some tests for removal of Food.getRevenue

kuromono committed:
Merge remote-tracking branch 'cube-upstream/master' into cube-fix-bug

kuromono committed:
Fix IntelliJ complaining: missing =

kuromono committed:
Merge pull request #280 from kuromono/cube-fix-bug

Update some tests for removal of Food.getRevenue

ZKathrynx committed:
checkstyle #2

ZKathrynx committed:
Merge branch 'master' into master

ZKathrynx committed:
Merge pull request #281 from ZKathrynx/master

checkstyle #2

parvathi14 committed:
Merge pull request #14 from AY1920S1-CS2113T-F09-2/master

Updating personal repo

Commit:
Merge branch 'master' of https://github.com/parvathi14/main

ZKathrynx committed:
checkstyle foodlist

ZKathrynx committed:
Merge branch 'master' of https://github.com/ZKathrynx/main

ZKathrynx committed:
Merge pull request #282 from ZKathrynx/master

Checkstyle foodlist

kuromono committed:
Update README

LL-Pengfei committed:
no message

LL-Pengfei committed:
no message

ZKathrynx committed:
finish checkstyle main

ZKathrynx committed:
Merge pull request #283 from ZKathrynx/master

finish checkstyle main

parvathi14 committed:
Merge pull request #15 from AY1920S1-CS2113T-F09-2/master

Updating personal repo

Commit:
Merge branch 'master' of https://github.com/parvathi14/main

ZKathrynx committed:
finish checkstyle

ZKathrynx committed:
Merge pull request #284 from ZKathrynx/master

finish checkstyle

ZKathrynx committed:
fixed repeated promotion bug

ZKathrynx committed:
Merge pull request #285 from ZKathrynx/master

fixed repeated promotion bug

LL-Pengfei committed:
no message

ZKathrynx committed:
fixed overviewdisplay product sum bug

ZKathrynx committed:
Merge pull request #286 from ZKathrynx/master

fixed overviewdisplay product sum bug

ZKathrynx committed:
sort type bug

ZKathrynx committed:
Merge pull request #287 from ZKathrynx/master

sort type bug

ZKathrynx committed:
fixed update bug

ZKathrynx committed:
Merge pull request #288 from ZKathrynx/master

fixed update bug

ZKathrynx committed:
added command JUnit test

ZKathrynx committed:
parser util JUnit test

ZKathrynx committed:
parser test #1

LL-Pengfei committed:
Merge pull request #289 from LL-Pengfei/master

javadoc comment cleaning

LL-Pengfei committed:
no message

ZKathrynx committed:
parser test

ZKathrynx committed:
Merge pull request #290 from ZKathrynx/master

JUnit test

LL-Pengfei committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-F09-2/main

ZKathrynx committed:
fixed reminder parser bug

ZKathrynx committed:
Merge pull request #291 from ZKathrynx/master

fixed reminder parser bug

LL-Pengfei committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-F09-2/main

LL-Pengfei committed:
new code added break the functionality (fail normal test) - remove

LL-Pengfei committed:
Merge pull request #292 from LL-Pengfei/master

bug fix

LL-Pengfei committed:
important bug fix for revenue

LL-Pengfei committed:
Merge pull request #293 from LL-Pengfei/master

important bug fix for revenue

kuromono committed:
Supports Add -c cost, Update -c cost

kuromono committed:
Merge pull request #294 from kuromono/cube-fix-bug

Add/Update support cost

kuromono committed:
Update command test for cost

kuromono committed:
Merge pull request #295 from kuromono/cube-fix-bug

Update command test for cost

LL-Pengfei committed:
reformat

LL-Pengfei committed:
Merge pull request #296 from LL-Pengfei/master

reformat

parvathi14 committed:
Merge pull request #16 from AY1920S1-CS2113T-F09-2/master

Updating person repo

Commit:
v1.4: Minor changes

Commit:
Merge branch 'master' of https://github.com/parvathi14/main

# Conflicts:
#	src/main/java/cube/logic/command/HelpCommand.java

Commit:
v1.4: Fixed Reminder Command Test file

parvathi14 committed:
Merge pull request #297 from parvathi14/master

v1.4: Fixed Reminder Command Test file

tygq13 committed:
bug fix in cost

tygq13 committed:
Merge pull request #298 from tygq13/bugfix

bug fix in cost

tygq13 committed:
remove food's revenue attribute

tygq13 committed:
Merge branch 'master' into bugfix

tygq13 committed:
Merge pull request #299 from tygq13/bugfix

remove food's revenue attribute

LL-Pengfei committed:
aesthemetic formatting

LL-Pengfei committed:
corrected the feedback message for profit command

LL-Pengfei committed:
Merge pull request #300 from LL-Pengfei/master

profit command feedback message corrected + aesthemetic changes

ZKathrynx committed:
Merge pull request #21 from AY1920S1-CS2113T-F09-2/master

Sync

ZKathrynx committed:
final check style

kuromono committed:
Moved OverviewDisplay update outside as some commands not updated overview

- e.g. batch

kuromono committed:
Update JavaDocs for GUI parts

kuromono committed:
Merge pull request #301 from kuromono/cube-fix-bug

Fix OverviewDisplay bug, added GUI JavaDocs

ZKathrynx committed:
another finalized checkstyle

ZKathrynx committed:
Merge branch 'master' into master

ZKathrynx committed:
Merge pull request #302 from ZKathrynx/master

Checkstyle

kuromono committed:
Remove FoodRevenue Tests

tygq13 committed:
Merge branch 'master' of https://github.com/tygq13/main

kuromono committed:
Remove the extra line in javadocs

tygq13 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-F09-2/main

tygq13 committed:
floating point bug in profit

tygq13 committed:
Merge pull request #305 from tygq13/bugfix

Bugfix in profit floating point

kuromono committed:
Merge remote-tracking branch 'cube-upstream/master' into cube-junit

kuromono committed:
Fix Checkstyle : ReminderCommandTest

tygq13 committed:
minor change in messages

tygq13 committed:
Merge pull request #306 from tygq13/bugfix

minor change in messages

ZKathrynx committed:
fixed default reminder bug

ZKathrynx committed:
Merge pull request #307 from ZKathrynx/master

fixed default reminder bug

kuromono committed:
Fix : Batch CSV import exception of invalid format

kuromono committed:
Merge remote-tracking branch 'cube-upstream/master' into cube-junit

kuromono committed:
Merge pull request #304 from kuromono/cube-junit

Fix Batch CSV Bug & Remove FoodRevenue Tests

ZKathrynx committed:
static message error

ZKathrynx committed:
Merge pull request #22 from AY1920S1-CS2113T-F09-2/master

Sync

ZKathrynx committed:
Merge branch 'master' of https://github.com/ZKathrynx/main

ZKathrynx committed:
Merge pull request #308 from ZKathrynx/master

static message error

kuromono committed:
Fixed static string repeating GUI bug : config

kuromono committed:
Change message success var name for checkstyle

kuromono committed:
Merge remote-tracking branch 'cube-upstream/master' into cube-junit

# Conflicts:
#	src/main/java/cube/logic/command/ReminderCommand.java fixed

kuromono committed:
Merge pull request #309 from kuromono/cube-junit

Fix Config having same issue GUI bug

tygq13 committed:
some author tags

tygq13 committed:
Merge pull request #310 from tygq13/bugfix

some author tags

kuromono committed:
Fix Integer Bug Test

LL-Pengfei committed:
aesthemetic changes

LL-Pengfei committed:
aesthemetic change

LL-Pengfei committed:
Merge pull request #312 from LL-Pengfei/master

javadoc comment - aesthemetic changes

LL-Pengfei committed:
aesthemetic change

LL-Pengfei committed:
Merge pull request #313 from LL-Pengfei/master

aesthemetic change - trivial

kuromono committed:
Modified ModelManager to support ProfitCommand Test, fix failing

kuromono committed:
Fix DeletePromotionCommandTest

kuromono committed:
Fix AddPromotionCommandTest

kuromono committed:
Fix partial : ReminderCOmmandTest

kuromono committed:
Merge pull request #311 from kuromono/cube-junit

Fixing JUnit tests

kuromono committed:
Add DG UG to GITHUB

kuromono committed:
Merge pull request #314 from kuromono/cube-junit

Add DG UG to GITHUB
